code/controllers/subsystem/dbcore.dm

```diff
@@ -0,0 +1,246 @@
+SUBSYSTEM_DEF(dbcore)
+	name = "Database"
+	flags = SS_NO_INIT|SS_NO_FIRE
+	init_order = INIT_ORDER_DBCORE
+	var/const/FAILED_DB_CONNECTION_CUTOFF = 5
+
+	var/const/Default_Cursor = 0
+	var/const/Client_Cursor = 1
+	var/const/Server_Cursor = 2
+	//conversions
+	var/const/TEXT_CONV = 1
+	var/const/RSC_FILE_CONV = 2
+	var/const/NUMBER_CONV = 3
+	//column flag values:
+	var/const/IS_NUMERIC = 1
+	var/const/IS_BINARY = 2
+	var/const/IS_NOT_NULL = 4
+	var/const/IS_PRIMARY_KEY = 8
+	var/const/IS_UNSIGNED = 16
+// TODO: Investigate more recent type additions and see if I can handle them. - Nadrew
+
+	var/_db_con// This variable contains a reference to the actual database connection.
+	var/failed_connections = 0
+
+/datum/controller/subsystem/dbcore/PreInit()
+	_db_con = _dm_db_new_con()
+
+/datum/controller/subsystem/dbcore/Recover()
+	_db_con = SSdbcore._db_con
+
+/datum/controller/subsystem/dbcore/Shutdown()
+	// This is as clsoe as we can get to the true round end before Disconnect() without changing where it's called, deafeating the reason this is a subsystem
+	if(SSdbcore.Connect())
+		var/sql_station_name = sanitizeSQL(station_name())
+		var/datum/DBQuery/query_round_end = SSdbcore.NewQuery("INSERT INTO [format_table_name("round")] (shutdown_datetime, game_mode_result, end_state, station_name) VALUES (Now(), '[ticker.mode_result]', '[ticker.end_state]', '[sql_station_name]') WHERE id = [GLOB.round_id]")
```

Comment:
Note to self. Make this more secure or in its own proc. Pretty much be sure this type of stuff is locked down.

Reply:
"More secure"? Only thing you might consider is adding more sanitizeSQL() on the entries, but there's not much else to do. Putting it in a different proc would only make it more vulnerable in general, really-- admins can arbitrarily call functions, they cannot do so with specific lines inside a proc.

Reply:
It was a concern over someone injecting into it before round end @tigercat2000 by lemons.